internal/tasks/process-file-updates-task/process-file-updates-task.yaml

```diff
@@ -243,7 +243,61 @@ spec:
         fi
 
         echo -e "\n*** START LOCAL CHANGES ***\n"
-        git diff
+        git diff | tee "${TEMP}"/tempMRFile.diff
+
+        if [[ -s "${TEMP}"/tempMRFile.diff ]]; then
+            # replacements exist
+            awk '/^@@/ {match($0, /@@ ([^@]+) @@/, arr); print arr[1]}' "${TEMP}"/tempMRFile.diff |\
+               tee "${TEMP}"/lineFile
+        fi
+
+        openMRList=$(glab mr list -R "${UPSTREAM_REPO}" --search "Konflux release" |grep "^!"| cut -f1 | tr -d '!')
+        for mrNum in ${openMRList} ; do
```

Comment:
It seems to me this whole logic is only for the use case that there is replacement to be done, but maybe one of the open MRs already contains it. That's definitely a valid use case.

But what if the change has already been merged? Then we will probably fail on line 235, saying there is nothing to change, right? But if you run the same release repeatedly, that use case shouldn't fail.

Now the question is how to handle this - we will need to separate these two cases:

- there is substitution to be done because the files already contain the new strings -> pass
- there is substitution to be done for some other reason, e.g. we didn't find the lines to be updated -> fail

Reply:
> But what if the change has already been merged? Then we will probably fail on line 235, saying there is nothing to change, right? But if you run the same release repeatedly, that use case shouldn't fail. How to differentiate the MR is merged and the file doesn't need to be changed at all?

Reply:
> Do we still need to check the merged PRs?

I guess that would be one way to handle it. But probably not necessary. It would be difficult to decide how far into the history we want to go.

Reply:
IMO, if the "goal string" aka the string you are trying to insert already exists in the proper file, exit 0. No need to check merged MRs. If an MR is open with the exact change already, say that (idempotent) and exit 0 (ideally printing the existing MR url). If the file to be changed doesn't exist, exit 0. I *think* this is in agreement with what Martin has been commenting

Reply:
If only the return value is changed from "Failed" to "Success" in the block of "line 235",  the test  case of "test-process-file-updates-replacements-error.yaml" fails.  I added a variable of "keyNotFound" to check if the key is correct.  And if the file is not valid, it's handled in line 157. Is that ok?

Reply:
I am reading through what the script does and I am not sure now. Until now, if a key is not found, that replacement will be skipped, but at the end we check that at least one replacement happened, otherwise we fail.

So yes, I think what you say sounds correct - we should fail if we cannot find the target file or the target key. But if those are found and there is no diff once the replacement is applied, it's still ok.

But I noticed there is also this `seed` option. It seems that one should be ok - if you run it again, it will just override the file with the seed string again, so there will be no diff. Previously this would fail on "no diff", but as we said, we will no longer fail on "no diff".